src/NetMQ/NetMQPoller.cs

```diff
@@ -225,6 +252,11 @@ public void Remove(ISocketPollable socket)
         {
             if (socket == null)
                 throw new ArgumentNullException(nameof(socket));
+
+            // JASells: not sure I agree with this thow.
+            // If trying to remove a disposed socket, why complain?  It *might* work.  The issue
+            // is if the poller's thread tries to actually service the socket before the remove call...
+
```

Comment:
I'd remove this from the code and discuss in an issue/PR instead where conversation can happen more easily.

The rationale here was to help people discover when they're tearing things down in the wrong order, as that is a common cause of deadlocks and other problems. If you're going to fail some of the time, it's better to fail all of the time as it makes it easier to write code correctly.

Reply:
Yes, this is a note to do that =)